src/extrinsic/offline_extrinsic.rs

```diff
@@ -19,22 +19,27 @@
 
 use crate::Api;
 use ac_compose_macros::compose_extrinsic_offline;
-use ac_primitives::{ExtrinsicParams, FrameSystemConfig, SignExtrinsic, UncheckedExtrinsicV4};
+use ac_primitives::{
+	config::Config, extrinsic_params::ExtrinsicParams, SignExtrinsic, UncheckedExtrinsicV4,
+};
 use codec::Encode;
 
-impl<Signer, Client, Params, Runtime> Api<Signer, Client, Params, Runtime>
-where
-	Signer: SignExtrinsic<Runtime::AccountId>,
-	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
-	Runtime: FrameSystemConfig,
-{
+type ExtrinsicAddress<T> =
+	<<T as Config>::ExtrinsicSigner as SignExtrinsic<<T as Config>::AccountId>>::ExtrinsicAddress;
+type Signature<T> =
```

Comment:
Can't you take the Signature directly from the Config? Respective, is that type defintion actually necessary? Would it not be possible to use
```rust
// ExtrinsicAddress
T::Address
// Signture
T::Signature
```

Don't think there's a shortcut for the SignedExtra though.

Reply:
Compiler requires these types' definition:

```
error[E0308]: arguments to this function are incorrect

--> src/extrinsic/offline_extrinsic.rs:44:20
   |
44 |             Some(signer) => compose_extrinsic_offline!(signer, call, self.extrinsic_params(nonce)),
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: expected `ac_primitives::Config::Address`, found `ac_primitives::SignExtrinsic::ExtrinsicAddress`

```

Reply:
Ah, that's a pity. But yeah, makes sense. Leave as is then